odcsapi-jetty-standalone/build.gradle

```diff
@@ -34,15 +35,53 @@ publishing {
     }
 }
 
-task startJetty(type: JavaExec) { ->
+task downloadOpenDcsInstaller(type: de.undercouch.gradle.tasks.download.Download) {
+    src 'https://github.com/opendcs/opendcs/releases/download/' + libs.versions.opendcs.get() +'/opendcs-installer-' + libs.versions.opendcs.get() + '.jar'
+    dest new File("${project.buildDir}/tmp/opendcs-installer.jar")
+    onlyIfModified true
+}
+
+task installOpenDcs(type: JavaExec) {
+    dependsOn downloadOpenDcsInstaller
+    mainClass = '-jar'
+    args = ["${project.buildDir}/tmp/opendcs-installer.jar", 'config/auto-installation-script.xml']
+}
+
+task copyUserProperties(type: Copy) {
+    from 'config'
+    include 'user.properties'
+    into "${project.buildDir}/opendcs"
+    filter(ReplaceTokens, tokens: ['postgresdb.url': project.findProperty('postgresdb.url') ?: ''])
+}
+
+task writeCredentials {
+    def username = project.findProperty("db.username")
+    def password =  project.findProperty("db.password")
+    def file = project.file("${project.buildDir}/opendcs/credentials.txt")
+    file.write("${username}\n${password}\n")
+}
+
+task writeUserAuthFile(type: JavaExec) { ->
+    dependsOn writeCredentials
+    classpath = sourceSets.test.runtimeClasspath
+    mainClass = "ilex.util.UserAuthFile"
```

Comment:
FYI opendcs 7.0.5+ have support for retrieving credentials from the environment

use AuthSource.getCredentials(<the db auth file string>);

See https://github.com/opendcs/opendcs/blob/2fc798892dc1d49840e579e75ff0ca43bc72c54d/src/test/java/org/opendcs/authentication/EnvironmentAuthSourceTest.java#L27 for a full examples

Reply:
Perfect, thanks!